### ISSUE TITLE
Implement connection force-close via ping timeout

### ISSUE THREAD

**Issue by andrewgodwin** [django/daphne#33]
We need to hook up an `onPong` handler to a time variable and then force-close the socket if we don't get a pong for a while (tie it to ping_interval somehow)

**Comment by globophobe** [comment]
I need this. Would something like this be enough?

# Force-close socket if pong not received, before next ping is sent
self.setProtocolOptions(autoPingTimeout=ping_interval-1)

**Comment by andrewgodwin** [comment]
It can just be worked into the existing ping sending code in Daphne - there's already a method that gets called every so often, just add a timeout check there.

**Comment by globophobe** [comment]
My first comment supposed there is a reason not to make use of setProtocolOptions for keepalive pings? Is it to save bytes, in case either server or client has sent a message recently? 

To accomplish the same, how about In HTTPFactory.__init__() 

```python
self.ws_factory.setProtocolOptions(autoPingTimeout=timeout)
```

Then, in WebSocketProtocol.check_ping() 

```python
_sendAutoPing()  # Instead of sendPing()
```

Looking at Autobahn's source, it seems that to force close via ping timeout for Daphne without Autobahn, most of Autobahn's relevant code would need to be duplicated in Daphne.

**Comment by andrewgodwin** [comment]
I didn't even realise Autobahn had this built-in - if it does, we should definitely reuse that rather than rolling it ourselves all over again.

**Comment by globophobe** [comment]
How adding the options for Autobahn's ```setProtocolOptions``` as an argparse subparser in cli.py, and pass them to ```HTTPFactory.init()``` as ```self.ws_factory.setProtocolOptions(**protocol_options)```? There are a number of useful options, such as ```allowedOrigins```, etc. 

If this change was made Daphne's ```WebsocketProtocol.last_data``` and ```Websocket.check_ping()``` wouldn't be needed at all. Just set Autobahn's ```autoPingInterval``` and ```autoPingTimeout```. 

Pings would be sent regardless of recent messages. However, with a 20 sec default, bytes sent over the network wouldn't increase substantially.

Happy to submit a pull request for this.

**Comment by andrewgodwin** [comment]
I'd say:
* Reuse the existing `ping_interval` CLI argument
* Make a new `ping_timeout` argument, default it to 1.5x the ping interval
* Pass them both to the factory and make it call `setProtocolOptions`

Pull request would be fantastic if you could find the time.

**Comment by andrewgodwin** [comment]
This was implemented in #34.
